src/main/scala/initial.scala

```diff
@@ -1,8 +1,29 @@
+import scala.annotation.tailrec
+import scala.concurrent.duration._
 
 object initial {
 
   def main(args: Array[String]): Unit = {
-    println("Hello")
+    var result = retry[Int](
+      block = () => 1+1,
+      acceptResult = res => res%2 == 0,
+      retries = List(0.seconds, 1.seconds, 2.seconds)
+    )
+    println(result)
+  }
+
+  def retry[A](block: () => A, acceptResult: A => Boolean, retries: List[FiniteDuration]): A = {
+    @tailrec
+    def retryT(block: () => A, acceptResult: A => Boolean, retries: List[FiniteDuration]): A = {
+      val blockerResult = block()
+      if (retries.isEmpty || acceptResult(blockerResult)) {
+        blockerResult
+      } else {
+        Thread.sleep(retries.head.toMillis)
+        retryT(block, acceptResult, retries.tail)
+      }
+    }
+    retryT(block=block, acceptResult=acceptResult, retries=retries)
```

Comment:
why you need internal retryT function? it has the same signature as external one, it is redundant.